### DISCUSSION TITLE
atomWithStorage "storage" listeners don't work + string values handling can cause issues

### DISCUSSION THREAD

**Discussion by leo-diehl** [pmndrs/jotai#2615] [Bug report]
From [the docs for the `storage` window event](https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event):
> Note: This won't work on the same [browsing context](https://developer.mozilla.org/en-US/docs/Glossary/Browsing_context) that is making the changes — it is really a way for other browsing contexts on the domain using the storage to sync any changes that are made.

I wonder if developers were aware about this limitation when the JSONStorage was implemented: https://github.com/pmndrs/jotai/blob/main/src/vanilla/utils/atomWithStorage.ts#L185
This creates a misleading notion that you can set localStorage values outside jotai and the values will be reflected on your atoms.

[Sandbox issue reproduction link](https://codesandbox.io/p/sandbox/busy-golick-28ykjn?file=%2Fsrc%2FApp.js%3A1%2C1-45%2C1&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clxbk3y7f00063b6le7uc0m2u%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clxbk3y7f00023b6lrw0mi9f9%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clxbk3y7f00033b6lbysu98dm%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clxbk3y7f00053b6lj6fu0ie6%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B56.31135734072022%252C43.68864265927978%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clxbk3y7f00023b6lrw0mi9f9%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxbk3y7f00013b6lndeekht9%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clxbk4enc00353b6l70cniibv%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A10%252C%2522startColumn%2522%253A30%252C%2522endLineNumber%2522%253A10%252C%2522endColumn%2522%253A30%257D%255D%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clxbl8bhd00023b6rcvxil6cx%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A1%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A45%252C%2522endColumn%2522%253A1%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.js%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clxbk3y7f00023b6lrw0mi9f9%2522%252C%2522activeTabId%2522%253A%2522clxbl8bhd00023b6rcvxil6cx%2522%257D%252C%2522clxbk3y7f00053b6lj6fu0ie6%2522%253A%257B%2522id%2522%253A%2522clxbk3y7f00053b6lj6fu0ie6%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clxbk3y7f00043b6lkg83qk9a%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clxbk3y7f00043b6lkg83qk9a%2522%257D%252C%2522clxbk3y7f00033b6lbysu98dm%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clxbk3y7f00033b6lbysu98dm%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

If this is known, I think we should at least add something in the docs to make it clear that all modification to the storage key must be done through jotai, otherwise the atom might misbehave.

**Comment by dai-shi** [comment]
I'm not exactly sure what you suggest. Can you spot the unexpected behavior?

Are you suggesting either a) fix an issue, or b) fix the docs?
